Engine/Common/Types.h

```diff
@@ -1432,80 +1904,69 @@ struct Float4 {
     {
         return btQuaternion(X, Y, Z, W);
     }
+
 #endif // LEVIATHAN_USING_OGRE
 
     // ----------------- Quaternions ------------------- //
     DLLEXPORT static inline Float4 CreateQuaternionFromAngles(const Float3& angles)
     {
         // multiplied by 0.5 to get double the value //
-        float cosx = cosf(0.5f * angles.X);
-        float cosy = cosf(0.5f * angles.Y);
-        float cosz = cosf(0.5f * angles.Z);
-
-        float sinx = sinf(0.5f * angles.X);
-        float siny = sinf(0.5f * angles.Y);
-        float sinz = sinf(0.5f * angles.Z);
+        const float cosx = cosf(0.5f * angles.X);
+        const float cosy = cosf(0.5f * angles.Y);
+        const float cosz = cosf(0.5f * angles.Z);
 
+        const float sinx = sinf(0.5f * angles.X);
+        const float siny = sinf(0.5f * angles.Y);
+        const float sinz = sinf(0.5f * angles.Z);
 
-        Float4 quaternion((Float4)0);
-        // compute quaternion //
-        quaternion.X = cosz * cosy * sinx - sinz * siny * cosx;
-        quaternion.Y = cosz * siny * cosx + sinz * cosy * sinx;
-        quaternion.Z = sinz * cosy * cosx - cosz * siny * sinx;
-        quaternion.W = cosz * cosy * cosx * sinz * siny * sinx;
-
-        return quaternion;
+        return Float4(
+            // compute quaternion //
+            cosz * cosy * sinx - sinz * siny * cosx, cosz * siny * cosx + sinz * cosy * sinx,
+            sinz * cosy * cosx - cosz * siny * sinx, cosz * cosy * cosx * sinz * siny * sinx);
```

Comment:
Would be nice to format this so that before each line was the component it computes:
```
// X
cosz * cosy * sinx - sinz * siny * cosx, 
// Y
cosz * siny * cosx + sinz * cosy * sinx,
// etc.
```